### PR TITLE
Fixed issue of Register button not changing after being clicked

### PR DESCRIPTION
**Pull request by Joyosmit**
**What kind of change does this PR introduce?**
Previously clicking the register button did not reflect any change on the ui.
Now the button is disabled after registering

**Issue Number:**
28

Fixes #28

**Snapshots/Videos:**

**Other information**
Kindly review

### CODE REVIEW

**Comment by Doraemon012** [comment]
@Joyosmit Please provide a screen recording of the final output.

**Comment by Joyosmit** [comment]
https://github.com/Tanay-ErrorCode/lupo-skill/assets/143820879/0783eba9-0f16-425c-b913-2f4d79515921

File: src/components/DashBoard/DashBoard.tsx [hunk 1]
```diff
@@ -8,171 +8,183 @@ import EventCard from "../Cards/EventCard/EventCard";
...
+                            eventList.push(event);
+                        }
+                    });
+                }
+
+                eventList.forEach((eventId: string) => {
+                    const trimmedEventId = eventId.trim();
+                    const eventsRef = ref(database, "events");
+                    
+                    if(trimmedEventId[0] =='{'){
```
Comment (Doraemon012) [inline src/components/DashBoard/DashBoard.tsx:90]: Please explain what does this part of code do.

File: src/components/DashBoard/DashBoard.tsx [hunk 2]
```diff
@@ -8,171 +8,183 @@ import EventCard from "../Cards/EventCard/EventCard";
...
-    }
-    const fetchData = async () => {
-      const usersRef = ref(database, "users");
-      const userEmailId = localStorage.getItem("userEmailId");
+    useEffect(() => {
+        if (localStorage.getItem("userEmailId") == null) {
+            window.location.href = "#/";
+            toast.warn("You are not signed in", { transition: Zoom });
+        }
+        console.log("BROOO",localStorage.getItem("userEmailId"))
```
Comment (Doraemon012) [inline src/components/DashBoard/DashBoard.tsx:52]: Remove your debug comments.

File: src/components/DashBoard/DashBoard.tsx [hunk 3]
```diff
@@ -8,171 +8,183 @@ import EventCard from "../Cards/EventCard/EventCard";
...
-          const trimmedEventId = eventId.trim();
-          const eventsRef = ref(database, "events");
-          const eventRef = child(eventsRef, trimmedEventId);
-          get(eventRef).then((snapshot) => {
             if (snapshot.exists()) {
-              const event = snapshot.val();
-              setEventCardsData((prev: any[]) => [...prev, event]);
-              setTotalPages(Math.ceil(eventCardsData.length / itemsPerPage));
-              setIsLoading(false);
+                console.log(snapshot)
```
Comment (Doraemon012) [inline src/components/DashBoard/DashBoard.tsx:68]: Remove this.

File: src/components/DashBoard/DashBoard.tsx [hunk 4]
```diff
@@ -8,171 +8,183 @@ import EventCard from "../Cards/EventCard/EventCard";
...
+                    }
+                    const eventRef = child(eventsRef, trimmedEventId);
+                    get(eventRef).then((snapshot) => {
+                        if (snapshot.exists()) {
+                            const event = snapshot.val();
+                            setEventCardsData((prev: any[]) => [...prev, event]);
+                            setTotalPages(Math.ceil(eventCardsData.length / itemsPerPage));
+                            setIsLoading(false);
+                        } else {
+                            console.log("No data available in A");
```
Comment (Doraemon012) [inline src/components/DashBoard/DashBoard.tsx:101]: Revert this, or clarify what is `A`.

File: src/components/DashBoard/DashBoard.tsx [hunk 5]
```diff
@@ -8,171 +8,183 @@ import EventCard from "../Cards/EventCard/EventCard";
...
+                        }
+                    });
+                });
+                setIsLoading(false);
             } else {
-              console.log("No data available");
+                console.log("No data available in B");
```
Comment (Doraemon012) [inline src/components/DashBoard/DashBoard.tsx:107]: Revert this, or clarify what is `B`.

File: src/components/ProfilePage/ProfilePage.tsx [hunk 6]
```diff
@@ -127,9 +128,10 @@ const ProfilePage = () => {
             const trimmedEventId = eventId.trim();
             const eventsRef = ref(database, "events");
             const eventRef = child(eventsRef, trimmedEventId);
-            get(eventRef).then((snapshot) => {
+            get(eventRef).then((snapshot: any) => {
               if (snapshot.exists()) {
                 const event = snapshot.val();
+                console.log("EVENT HEHE ", event)
```
Comment (Doraemon012) [inline src/components/ProfilePage/ProfilePage.tsx:134]: Remove.

File: src/components/ProfilePage/ProfilePage.tsx [hunk 7]
```diff
@@ -151,6 +153,9 @@ const ProfilePage = () => {
         eventList.forEach((eventId: string) => {
           const trimmedEventId = eventId.trim();
           const eventsRef = ref(database, "events");
+          if (trimmedEventId[0] == '{') {
```
Comment (Doraemon012) [inline src/components/ProfilePage/ProfilePage.tsx:156]: Explain this part of code. What is the reason behind adding this ?

**Review by Doraemon012 — CHANGES_REQUESTED** [review]
@Joyosmit Please see the requested changes.

File: src/components/DashBoard/DashBoard.tsx (hunk 1 above)
Comment (Joyosmit) [reply src/components/DashBoard/DashBoard.tsx:90]: In the previous lines,
if (snapshot.hasChild("registeredEvents")) {
                    const registeredEvents = snapshot.val().registeredEvents.split(",");
                    registeredEvents.forEach((event: string) => {
                        if (!eventList.includes(event)) {
                            eventList.push(event);
                        }
                    });
                }
the string "{card.id}" was getting pushed into eventList and hence, giving error in the child function as '{' is not allowed.
That's why i use this to ignore this case

File: src/components/ProfilePage/ProfilePage.tsx (hunk 7 above)
Comment (Joyosmit) [reply src/components/ProfilePage/ProfilePage.tsx:156]: For some reason, eventList was having a item '{card.id}' in it which was giving error with child function. So i ignored that case

File: src/components/ProfilePage/ProfilePage.tsx (hunk 6 above)
Comment (Joyosmit) [reply src/components/ProfilePage/ProfilePage.tsx]: Removed

File: src/components/DashBoard/DashBoard.tsx (hunk 2 above)
Comment (Joyosmit) [reply src/components/DashBoard/DashBoard.tsx]: removed

File: src/components/DashBoard/DashBoard.tsx (hunk 3 above)
Comment (Joyosmit) [reply src/components/DashBoard/DashBoard.tsx]: removed

File: src/components/DashBoard/DashBoard.tsx (hunk 4 above)
Comment (Joyosmit) [reply src/components/DashBoard/DashBoard.tsx]: reverted

File: src/components/DashBoard/DashBoard.tsx (hunk 5 above)
Comment (Joyosmit) [reply src/components/DashBoard/DashBoard.tsx]: reverted

File: src/components/DashBoard/DashBoard.tsx (hunk 1 above)
Comment (Tanay-ErrorCode) [reply src/components/DashBoard/DashBoard.tsx:89]: This check is unnecessary, remove it.

This is the `eventList` array

image

File: src/components/ProfilePage/ProfilePage.tsx (hunk 7 above)
Comment (Tanay-ErrorCode) [reply src/components/ProfilePage/ProfilePage.tsx:156]: Remove ,this check is unnecessary.

**Comment by Tanay-ErrorCode** [comment]
@Joyosmit Please do the required changes.

**Review by Doraemon012 — APPROVED** [review]

**Comment by Doraemon012** [comment]
@Joyosmit Please update your branch, it has been outdated.

image

https://github.com/Tanay-ErrorCode/lupo-skill/assets/116624667/6fe638ca-bc95-4a5e-a43a-11d7757f21dd

**Comment by Joyosmit** [comment]
Alright, thx. I have updated my branch as per your instruction.

**Review by Doraemon012 — APPROVED** [review]

**Review by Doraemon012 — APPROVED** [review]

**Review by Tanay-ErrorCode — APPROVED** [review]

**Pull request merged**